src/olympia/devhub/helpers.py

```diff
@@ -115,7 +115,17 @@ def docs_breadcrumbs(context, items=None):
 @jinja2.contextfunction
 def add_file_modal(context, title, action, action_label, modal_type='file'):
     addon = context['addon']
-    upload_url = reverse('devhub.upload_for_addon', args=[addon.slug])
+    version = context.get('version',
+                          addon.find_latest_version_including_rejected())
+    if version:
+        channel_id = version.channel
+    else:
+        # short term fix - this function won't be used after new file upload.
+        channel_id = (amo.RELEASE_CHANNEL_LISTED if addon.is_listed else
+                      amo.RELEASE_CHANNEL_UNLISTED)
+    channel = amo.CHANNEL_CHOICES_API[channel_id]
```

Comment:
I've used it in a few places (and the companion `CHANNEL_CHOICES_LOOKUP`) as a way of converting between a constant and the string we might use in a url... is it that bad?  Is it the naming?